### DISCUSSION TITLE
Struggling with host network mode

### DISCUSSION THREAD

**Discussion by Foxlider** [wolveix/satisfactory-server#337] [Q&A] [answered]
I was having a lot of performance issues lately with some users being randomly kicked from the server, rubberbanding etc...

I have come to the hypothesis that the bridged network could be an issue. 

Since Docker Desktop 4.34 is out, Host networking is supported but while no error is being shown, I can't seem to see my server, even on local network. But it works on bridged net

Anyone got a similar issue or a solution ?

**Comment by wolveix** [comment]
@Foxlider I think your assessment is wrong. From helping many people on the official Satisfactory Discord server, these issues are almost always caused by low single-core CPU performance on the server, or low RAM allocation.

Docker's networking will not be the source of the issue.

**Reply by Foxlider** [reply]
The server's CPU is an i7-12700k currently running at 4GHz

My WSL2 config allows for 4 cores
```
[wsl2]
memory=16GB   # Limits VM memory in WSL 2 up to 16GB  WAS 12
processors=4 # Makes the WSL 2 VM use 4 virtual processors
```

The docker compose also limits to 16Gb
```yaml
deploy:
  resources:
    limits:
      memory: 12G
    reservations:
      memory: 8G
```

Are my settings wrong ? CPU not good enough ?

**Reply by wolveix** [reply]
@Foxlider the clock rate doesn't really matter, nor do the number of cores (this is a game server, after all). With that being said, yes your CPU's single-core performance seems sufficient.

With that said, it's hard to diagnose the issue further without knowing the rest of your environment (see the issue template for this repo). Ensure that you're not binding Satisfactory to a Windows directory, as WSL2's cross-system IO is very poor.

You mention multiple unrelated issues in your opening post, so it's a little hard to know how to help you exactly. I wouldn't bother changing the container's networking, that's not where the issue is after all. I also wouldn't use host networking where it can be avoided. Docker for Desktop uniquely introduces a number of challenges too. As you can see, there are 101 different things that _could_ be contributing to your issue :) I'd recommend asking for further help in the official Satisfactory Discord, and ensure that you supply all of your environment information, otherwise people will not be able to help you

**Reply by Foxlider** [reply]
> Ensure that you're not binding Satisfactory to a Windows directory, as WSL2's cross-system IO is very poor.

Ah... That might be it... 

I'll try that and then if it still fucks up, I'll check on the discord. 
How would you suggest to fix it ? Volume or can I still bind it to a folder ?
This is the current one 
```yaml
volumes:
    - 'C:\Users\Username\Documents\Tools\Docker\Satisfactory\config:/config'
```

**Reply by wolveix** [reply]
You can absolutely still bind it, but ensure that the bound directory lives within WSL2 (under your Linux distro), and not on your Windows system :)

**Reply by Foxlider** [reply]
I'm not entirely sure how to do that

**Reply by wolveix** [reply]
Under your Linux distro, run `echo $HOME` to find your home directory, then use that in place of `/home/your-user` below:

```
volumes:
    - '/home/your-user/satisfactory:/config'
```

**Reply by Foxlider** [reply]
I'm not using WSL Integration. It uses `docker-desktop` and `docker-desktop-data`. Should I ?

**Reply by wolveix** [reply, marked as answer]
@Foxlider in Docker for Desktop's settings, there's an option for it to utilize WSL2. In my experience, this is significantly more stable :)
